workflows/genomeqc.nf

```diff
@@ -26,15 +34,93 @@ workflow GENOMEQC {
 
     ch_versions = Channel.empty()
     ch_multiqc_files = Channel.empty()
+    ch_tree_data = Channel.empty()
+
+    ch_samplesheet
+        .map {
+            validateInputSamplesheet(it) // Input validation (check local subworkflow)
+        }
+        .branch {
+            ncbi: it.size() == 2
+            local: it.size() == 3
+        }
+        .set { ch_input }
 
     //
-    // MODULE: Run FastQC
+    // MODULE: Run create_path
     //
-    FASTQC (
-        ch_samplesheet
+
+    CREATE_PATH (
+        ch_input.ncbi
+    )
+
+    //
+    // MODULE: Run ncbigenomedownlaod
+    //
+
+    NCBIGENOMEDOWNLOAD ( 
+        CREATE_PATH.out.meta,
+        CREATE_PATH.out.accession,
+        [],
+        params.groups
     )
-    ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]})
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+    ch_versions = ch_versions.mix(NCBIGENOMEDOWNLOAD.out.versions.first())
+    
+    //
+    // Define gff and fasta varliables
+    //
+
+    fasta = NCBIGENOMEDOWNLOAD.out.fna.mix( ch_input.local.map { [it[0],file(it[2])] } )
+    gff   = NCBIGENOMEDOWNLOAD.out.gff.mix( ch_input.local.map { [it[0],file(it[1])] } )
+    
+    // Uncompress files if necessary | Consider using brances as an alternative
+
+    if (fasta.map { it[1].endsWith(".gz") } ) {
```

Comment:
This doesn't do what you think it does. You cannot use channels in an `if`. You should use either `branch` or `filter` here.

Reply:
I think this was fixed by @LaurenHuet in #72. I will incorporate this fix in the next merge.